### PR TITLE
lint & format `weights/*.py` part 8

### PR DESCRIPTION
**Pull request by jGaboardi**
This PR:
* formats and lints `weights/*.py` –  and should be part 8 of 8
* nearly completes the linting of `libpysal` -> there seem to be some loose ends in `cg` that I will take care of in a follow PR.

### CODE REVIEW

File: libpysal/weights/tests/test_util.py [hunk 1]
```diff
@@ -162,9 +162,9 @@ def test_higher_order_classes(self):
         util.higher_order(wsparse, 2)
         util.higher_order(ww, 2)
         ww.transform = "r"
-        wsparse_notbinary = wrook.sparse
+        wsparse_notbinary = wrook.sparse  # noqa F841
```
Comment (jGaboardi) [inline libpysal/weights/tests/test_util.py:165]: After `ruff` notified that this line is not being used, I looked a bit deeper and found that here are merely testing things don't fail (except the last one that I move out). Not sure if we want to do the same for the `wsparse_notbinary = wrook.sparse` or just remove it? Maybe outside the scope of this PR?

Comment (martinfleis) [reply libpysal/weights/tests/test_util.py:165]: We can keep it but don't assign to a variable if the result is not tested.

File: libpysal/weights/tests/test_util.py [hunk 2]
```diff
@@ -162,9 +162,9 @@ def test_higher_order_classes(self):
...
+        util.higher_order(wsparse, 2)
```
Comment (martinfleis) [inline libpysal/weights/tests/test_util.py:166]: Is this correctly moved outside of the pytest.raises?

File: libpysal/weights/tests/test_util.py [hunk 3]
```diff
@@ -200,7 +200,6 @@ def test_full2_w(self):
         ids = ["myID0", "myID1", "myID2", "myID3"]
         w = util.full2W(a, ids=ids)
         np.testing.assert_array_equal(w.full()[0], a)
-        w.full()[0] == a
```
Comment (martinfleis) [inline libpysal/weights/tests/test_util.py:203]: Why is this removed?

File: libpysal/weights/tests/test_util.py (hunk 2 above)
Comment (jGaboardi) [reply libpysal/weights/tests/test_util.py:166]: It does not raise. If it was there because it previously raised, it no longer does. But my guess is that it simply was indented by mistake.

File: libpysal/weights/tests/test_util.py (hunk 3 above)
Comment (jGaboardi) [reply libpysal/weights/tests/test_util.py:203]: The line above it actually does the testing. That line is a "pointless expression" according to `ruff` since it merely calls a comparison but does't actually assert anything.

**Review by martinfleis — APPROVED** [review]
